addons/sale/models/sale_order.py

```diff
@@ -743,6 +743,14 @@ def _onchange_company_id_warning(self):
                 }
             }
 
+    @api.onchange('company_id')
+    def _onchange_company_id(self):
```

Comment:
There is already an `_onchange_company_id` on the SO model in `sale_management`, so your method is 'nearly' never executed.